PWGHF/D2H/TableProducer/dataCreatorDV0Reduced.cxx

```diff
@@ -0,0 +1,184 @@
+    registry.add("hPtV0", "V0 candidates;V0 candidate #it{p}_{T} (GeV/#it{c});entries", {HistType::kTH1F, {{100, 0., 10.}}});
+    registry.add("hMassDs1", "Ds1 candidates;Ds1 candidate #inv. mass (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{100, 2., 3.}}});
+  }
+
+  void runDataCreation(aod::Collision const& collision,
+                       CandsDFiltered const& candsD,
+                       aod::V0Datas const& V0s)
+  {
+    // helpers for ReducedTables filling
+    int indexHfReducedCollision = lfReducedCollision.lastIndex() + 1;
+    // std::map where the key is the V0.globalIndex() and
+    // the value is the V0 index in the table of the selected v0s
+    std::map<int64_t, int64_t> selectedV0s;
+    bool fillHfReducedCollision = false;
+
+    for (const auto& candD : candsD) {
+      bool fillHfCandD = false;
+      float invMassD = hfHelper.invMassDplusToPiKPi(candD);
+      std::array<float, 3> pVecD = candD.pVector();
+
+      for (const auto& v0 : V0s) {
+        std::array<float, 3> pVecV0 = {v0.pxpos() + v0.pxneg(), v0.pypos() + v0.pyneg(), v0.pzpos() + v0.pzneg()};
+        float ptV0 = sqrt(pVecV0[0] * pVecV0[0] + pVecV0[1] * pVecV0[1]);
+        auto invMass2DV0 = RecoDecay::m2(std::array{pVecD, pVecV0}, std::array{MassDPlus, MassK0Short});
+        registry.fill(HIST("hMassDs1"), sqrt(invMass2DV0));
+
+        // fill V0 table
+        // if information on V0 already stored, go to next V0
+        if (!selectedV0s.count(v0.globalIndex())) {
+          hfCandV0(v0.posTrackId(), v0.negTrackId(),
+                   indexHfReducedCollision,
+                   v0.x(), v0.y(), v0.z(),
+                   v0.mK0Short(),
+                   pVecV0[0], pVecV0[1], pVecV0[2]);
+          selectedV0s[v0.globalIndex()] = hfCandV0.lastIndex();
+          registry.fill(HIST("hPtV0"), ptV0);
+          registry.fill(HIST("hMassV0"), v0.mK0Short());
+        }
+        fillHfCandD = true;
+      } // V0 loop
+
+      if (fillHfCandD) { // fill candDplus table only once per D candidate, only if at least one V0 is found
+        hfCandD(candD.prong0Id(), candD.prong1Id(), candD.prong2Id(),
+                indexHfReducedCollision,
+                candD.xSecondaryVertex(), candD.ySecondaryVertex(), candD.zSecondaryVertex(),
+                invMassD,
+                pVecD[0], pVecD[1], pVecD[2]);
+        fillHfReducedCollision = true;
+        registry.fill(HIST("hMassDToPiKPi"), invMassD);
+        registry.fill(HIST("hPtD"), candD.pt());
+      }
+    } // candsD loop
+    registry.fill(HIST("hEvents"), 1 + Event::Processed);
+    if (!fillHfReducedCollision) {
+      registry.fill(HIST("hEvents"), 1 + Event::NoDV0Selected);
+      return;
+    }
+    registry.fill(HIST("hEvents"), 1 + Event::DV0Selected);
+    // fill collision table if it contains a DPi pair a minima
+    lfReducedCollision(collision.posX(), collision.posY(), collision.posZ());
```

Comment:
Instead of using the one of LF that might change in time and create conflicts with our code, I would use the `HfRedCollisions` here. I will then in a second PR separate the cov matrix elements in a joinable table used only for the B mesons.

Reply:
Thanks, I'm reverting to HfRedCollisions for the time being